model/DataObject.php

```diff
@@ -1902,13 +1898,13 @@ public function belongsToComponent($component, $classOnly = true) {
 	}
 
 	/**
-	 * Return all of the database fields defined in self::$db and all the parent classes.
-	 * Doesn't include any fields specified by self::$has_one.  Use $this->hasOne() to get these fields
+	 * Return all of the database fields in this object
 	 *
 	 * @param string $fieldName Limit the output to a specific field name
+	 * @param string $table Out parameter of the table this db field is set to
 	 * @return array The database fields
 	 */
-	public function db($fieldName = null) {
+	public function db($fieldName = null, &$table = null) {
```

Comment:
I have the same concerns than the `&$compositeFields` comments earlier on - using a pass-by-reference parameter as a second return type waters down the API and makes it more prone to side effects. The one use of this parameter (`dbObject()`) could use an internal cache which is also created through `cache_database_fields()`? Admittedly hacky in its own way. And we really don't want to repeat this loop, given how often `dbObject()` is used throughout the class. Any smart ideas on how to avoid this reference?